azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1224,10 +1221,12 @@ private String findValueForJobVariable(final ExecutableNode node, final String j
 
     final Props outputProps = target.getOutputProps();
     if (outputProps == null) {
-      throw new IllegalStateException("No output properties were generated for job " + jobName + ".");
+      this.logger.warn("No output properties were generated for job " + jobName + ".");
+      return null;
     }
     if (!outputProps.containsKey(variable)) {
-      throw new IllegalStateException("No variable named " + variable + " found in job " + jobName + " output parameters. Properties: " + outputProps);
+      this.logger.warn("No variable named " + variable + " found in job " + jobName + " output parameters. Properties: " + outputProps);
```

Comment:
Clarification: what is the implication of this change? Currently, when the user code hasn't passed a property we throw an exception and it surfaces in the logs. How are we planning to have the same behavior after this fix?

Reply:
Good point!
After the fix, user can still see that message in the flow log but it will not KILL the flow directly but fail the condition check for the conditional job.